aerogear/mobile-security-service
# feat(git-hook): Run unit tests on pre-commit hook (AEROGEAR-8495)

## Motivation

This is an improvement to [AEROGEAR-8495](https://issues.jboss.org/browse/AEROGEAR-8495).

## What

Add automatic unit testing to the pre-commit hook.

## Why

To ensure all unit tests pass before pushing to remote.

## Verification Steps
 
**If you haven't already, check out the steps in https://github.com/aerogear/mobile-security-service/pull/24 to test the earlier versions of this hook**.

1. To test if the unit testing section of this hook runs correctly and blocks your commit, temporarily make a change to one of the unit tests to make it fail.
2. Run `git add . && commit --amend`
3. The commit should be prevented and you should see the following output:

```sh
...
Unit test(s) failed. Please fix your code and try again.
```

## Checklist:


## Progress
